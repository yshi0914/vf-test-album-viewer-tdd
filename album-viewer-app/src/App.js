import React from "react";
import "./App.css";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import Album from "./pages/Album";
import Photo from "./pages/Photo";
import FullPhoto from "./pages/FullPhoto";

function App() {
  return (
    <div className="App">
      <Router>
        <Switch>
          <Route exact path="/" component={Album} />
          <Route exact path="/photo/:id" component={Photo} />
          <Route path="/photo/full/:id" component={FullPhoto} />
        </Switch>
      </Router>
    </div>
  );
}

export default App;
