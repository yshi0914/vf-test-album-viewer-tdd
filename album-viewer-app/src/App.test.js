import React from "react";
import App from "./App";
import { shallow, mount } from "enzyme";

describe("App component after mounting", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<App />);
  });
  test("render the big title of this Album Viewer App", () => {
    expect(wrapper.find("h1").text()).toContain(
      "All albums with its associated User Name"
    );
  });
});

describe("App component snapshot testing", () => {
  test("matches the snapshot", () => {
    const component = shallow(<App />);
    expect(component.getElements()).toMatchSnapshot();
  });
});
