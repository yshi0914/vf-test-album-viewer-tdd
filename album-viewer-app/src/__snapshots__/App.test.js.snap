// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot testing matches the snapshot 1`] = `
Array [
  <div
    className="App"
  >
    <BrowserRouter>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/photo/:id"
        />
        <Route
          component={[Function]}
          path="/photo/full/:id"
        />
      </Switch>
    </BrowserRouter>
  </div>,
]
`;
